src/main/java/com/clivelee/crawler/EmailCrawler.java

```diff
@@ -0,0 +1,49 @@
+package com.clivelee.crawler;
+
+import com.clivelee.utils.TextFinder;
+import edu.uci.ics.crawler4j.crawler.Page;
+import edu.uci.ics.crawler4j.crawler.WebCrawler;
+import edu.uci.ics.crawler4j.parser.HtmlParseData;
+import edu.uci.ics.crawler4j.url.WebURL;
+
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.regex.Pattern;
+
+final class EmailCrawler extends WebCrawler {
+    private final static Pattern FILTERS = Pattern.compile(".*(\\.(css|js|html))$");
```

Comment:
Why CSS and JavaScript in the pattern?  The code for visit seems to restrict the email search to just HTML documents.